src/material/core/style/_elevation.scss

```diff
@@ -43,25 +43,30 @@ $prefix: 'mat-elevation-z';
   @else {
     // Copied from @material/elevation/_elevation-theme.scss#_box-shadow
     // TODO(mmalerba): Add support for graceful handling of CSS var color to MDC.
-    $shadow-color:
-      if(meta.type-of($color) == color and $opacity != null, rgba($color, $opacity), $color);
     $umbra-z-value: map.get(mdc-elevation.$umbra-map, $zValue);
     $penumbra-z-value: map.get(mdc-elevation.$penumbra-map, $zValue);
     $ambient-z-value: map.get(mdc-elevation.$ambient-map, $zValue);
+
+    $color-opacity: if($opacity != null, $opacity, 1);
+    $umbra-color: rgba($color, mdc-elevation.$umbra-opacity * $color-opacity);
```

Comment:
I don't follow why it's multiplying the opacities here. Doesn't this mean that if `mdc-elevation.$umbra-opacity` is hardcoded to 0.5, Material users won't be able to get it any higher?

Reply:
This should actually match the old behavior, https://github.com/angular/components/commit/7e94186fd5f7f6a94d56c1b55a9c125f51bf9804#diff-5dbefded555d45a2c2f55b6d0cd7b3c6c55dd0893745f9a87e1d5c611b5de292L38.Or am I missing out on something?

Reply:
Ah, I see.

Reply:
The only thing I was sondering is if the `if(meta.type-of($color) == color)` check is also needed here.

Like:
```scss
$umbra-color: if(meta.type-of($color) == color, rgba($color, mdc-elevation.$umbra-opacity * $color-opacity), $color)
```

Reply:
Even though we don't officially support CSS variables for theming yet, there are some apps that pass in variables in the palette anyway. We have the check in there to avoid compilation errors.

Reply:
@skrtheboss I ran this against our internal tests and it looks like these checks definitely need to be reintroduced, because they're used in some internal apps as well.